homeassistant/components/growatt_server/config_flow.py

```diff
@@ -86,7 +89,8 @@ async def async_step_plant(self, user_input=None):
 
             return self.async_show_form(step_id="plant", data_schema=data_schema)
 
-        if user_input is None and len(plant_info["data"]) == 1:
+        if user_input is None:
+            # single plant => mark it as selected
```

Comment:
The previous logic confused mypy, as it felt that `user_input` could still be `None`.
Since there are two guards above for empty data and for multiple data, I think we can remove the check here.